training accuracy reported as 0.0 with CUB hierarchical label for the classification

<!--
    Thank you very much for contributing to this project by creating an issue!
-->

**Describe the bug**
Training accuracy reported as 0.0 with CUB hierarchical label for the classification
here is the part of regression_config.json for this testing
![image](https://github.com/openvinotoolkit/training_extensions/assets/5232387/ca59d79e-269c-49f0-9e03-1817a15175e1)


<!--
    A clear and concise description of what the bug is.
    How has this issue affected you?
    What are you trying to accomplish?
-->

**Steps to Reproduce**

1. run classification regression TC 
```
$ CI_DATA_ROOT=/home/yunchule/data/ci_datasets/ tox -vvv -e tests-cls-py310-pt1 -- tests/regression/classification/test_classification.py::TestRegressionHierarchicalLabelClassification::test_otx_train[Custom_Image_Classification_EfficientNet-V2-S]
```
2. check the log
```
>       assert test_result["passed"] is True, test_result["log"]
E       AssertionError: [EfficientNet-V2-S] Performance: (0.0) < Criteria: (0.5418), threshold: 0.1.
E       assert False is True
```

**Environment:**

- OS: Ubuntu 20.04
- Framework version: Pytorch
- Python version: 3.10
- OpenVINO version: 2023.0
- CUDA/cuDNN version: 11.7
- GPU model and memory: 3090/24G

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,7 @@ All notable changes to this project will be documented in this file.
 - Fix h-label loss normalization issue w/ exclusive label group of singe label (<https://github.com/openvinotoolkit/training_extensions/pull/2604>)
 - Fix division by zero in class incremental learning for classification (<https://github.com/openvinotoolkit/training_extensions/pull/2606>)
 - Fix saliency maps calculation issue for detection models (<https://github.com/openvinotoolkit/training_extensions/pull/2609>)
+- Fix h-label bug of missing parent labels in output (<https://github.com/openvinotoolkit/training_extensions/pull/2626>)
 
 ## \[v1.4.3\]
 

--- a/src/otx/algorithms/classification/adapters/mmcls/datasets/otx_datasets.py
+++ b/src/otx/algorithms/classification/adapters/mmcls/datasets/otx_datasets.py
@@ -300,6 +300,7 @@ class OTXHierarchicalClsDataset(OTXMultilabelClsDataset):
 
     def __init__(self, **kwargs):
         self.hierarchical_info = kwargs.pop("hierarchical_info", None)
+        self.label_schema = kwargs.pop("label_schema", None)
         super().__init__(**kwargs)
 
     def load_annotations(self):
@@ -308,6 +309,13 @@ class OTXHierarchicalClsDataset(OTXMultilabelClsDataset):
         for i, _ in enumerate(self.otx_dataset):
             class_indices = []
             item_labels = self.otx_dataset[i].get_roi_labels(self.labels, include_empty=include_empty)
+            if self.label_schema:
+                # NOTE: Parent labels might be missing in annotations.
+                # This code fills the gap just in case.
+                full_item_labels = set()
+                for label in item_labels:
+                    full_item_labels.update(self.label_schema.get_ancestors(label))
+                item_labels = full_item_labels
             ignored_labels = self.otx_dataset[i].ignored_labels
             if item_labels:
                 num_cls_heads = self.hierarchical_info["num_multiclass_heads"]

--- a/src/otx/algorithms/classification/adapters/mmcls/task.py
+++ b/src/otx/algorithms/classification/adapters/mmcls/task.py
@@ -190,6 +190,7 @@ class MMClassificationTask(OTXClassificationTask):
         elif self._hierarchical:
             options_for_patch_datasets["type"] = "OTXHierarchicalClsDataset"
             options_for_patch_datasets["hierarchical_info"] = self._hierarchical_info
+            options_for_patch_datasets["label_schema"] = self._task_environment.label_schema
             options_for_patch_evaluation["task"] = "hierarchical"
         elif self._selfsl:
             options_for_patch_datasets["type"] = "SelfSLDataset"

--- a/tests/assets/datumaro_h-label/annotations/train.json
+++ b/tests/assets/datumaro_h-label/annotations/train.json
@@ -4,6 +4,11 @@
     "label": {
       "label_groups": [
         {
+          "name": "shape",
+          "group_type": "exclusive",
+          "labels": ["blue", "green"]
+        },
+        {
           "name": "blue",
           "group_type": "exclusive",
           "labels": ["blue_rectangle", "blue_circle", "blue_triangle"]

--- a/tests/assets/datumaro_h-label/annotations/valid.json
+++ b/tests/assets/datumaro_h-label/annotations/valid.json
@@ -4,6 +4,11 @@
     "label": {
       "label_groups": [
         {
+          "name": "shape",
+          "group_type": "exclusive",
+          "labels": ["blue", "green"]
+        },
+        {
           "name": "blue",
           "group_type": "exclusive",
           "labels": ["blue_rectangle", "blue_circle", "blue_triangle"]

--- a/tests/assets/datumaro_h-label_class_decremental/annotations/train.json
+++ b/tests/assets/datumaro_h-label_class_decremental/annotations/train.json
@@ -4,6 +4,11 @@
     "label": {
       "label_groups": [
         {
+          "name": "shape",
+          "group_type": "exclusive",
+          "labels": ["blue", "green"]
+        },
+        {
           "name": "blue",
           "group_type": "exclusive",
           "labels": ["blue_rectangle", "blue_circle", "blue_triangle"]

--- a/tests/assets/datumaro_h-label_class_decremental/annotations/valid.json
+++ b/tests/assets/datumaro_h-label_class_decremental/annotations/valid.json
@@ -4,6 +4,11 @@
     "label": {
       "label_groups": [
         {
+          "name": "shape",
+          "group_type": "exclusive",
+          "labels": ["blue", "green"]
+        },
+        {
           "name": "blue",
           "group_type": "exclusive",
           "labels": ["blue_rectangle", "blue_circle", "blue_triangle"]

